src/pages/guide/components/Carousel/Carousel.tsx

```diff
@@ -1,6 +1,7 @@
-import { ReactNode, useRef, useState } from 'react';
+import { ReactNode } from 'react';
 import clsx from 'clsx';
-import debounce from 'lodash/debounce';
+import useEmblaCarousel from 'embla-carousel-react';
```

Comment:
the reason why i went with embla carousel is due to its popularity, how small the bundle size is for this carousel library, and how easy it is to implement. 

I had to refactor how we did the carousel because of the user could skip past each slide within the carousel, but we should only allow the user to swipe one at a time. It was quite difficult to figure out how I could prevent the user from swiping/scrolling past the next slide item, so instead, it is better to go with a library to handle this, rather than trying to do it from scratch.

Wallets team have also implemented embla in their project, and we can also implement this library in the deriv-com/ui library.

Reply:
Is there a specific reason why you chose not to implement it in the deriv-com/ui library?

Reply:
If i were to upgrade the deriv-com/ui library to the latest version, theres quite a few things that would break in the UI 😅 so for the time being I decided to implement it from our side

Reply:
Why is that breaking the UI?

Reply:
@ameerul-deriv  but why are we restricting the user from skipping the slide? i think its an expected behaviour of carousel that the user can skip and click on the dots to directly access the slide.

Reply:
yes its fine if you click on the dots, but when the user swipes, the user can swipe a little bit, but it slides all the way til the end of the carousel, which isn't a good UX

Reply:
but generally for carousels, its supposed to be one at a time

Reply:
https://github.com/user-attachments/assets/d4b6332a-668c-4f92-af20-ca3d515145e8

i think its expected..

Reply:
https://github.com/user-attachments/assets/ef74f728-4e2a-4ff9-9cb0-0409cf00f6cc

@ameerul-deriv  do u mean the swiping from first to last?

Reply:
yeah, the swipe shouldn't be too much, on mobile, its quite sensitive, and its better to swipe through each item one at a time so users can see each slide. they have the option to skip items using the dot buttons

https://github.com/user-attachments/assets/1ea4e023-3fcf-4f65-a5e5-c6a71e589c64

Reply:
why are we using a library for this?

Reply:
as I have mentioned previously, to achieve sliding one item at a item without it swiping all the way to the end, i had to go with embla. I had tried to do numerous variations without a library and did the calculations from my side, but there were too many glitches. 

When you're on responsive, because the swiping is quite sensitive, it can swipe past a few items in the carousel. Now imagine we have a carousel thats even larger in the future with even more items in the list. This also caused bugs for the dot buttons, as it couldn't calculate which slide it is viewing. Hence why i opted for embla.

1. It is a super small library thats focused purely on carousels
2. I had discussed with wallets team about it as they had used it in their project
3. In terms of UX this is the better user experience. The action of swiping should be one by one so users can see the content individually rather than slowly adjusting the carousel. If they want to skip ahead, thats where the Dot buttons come in where they can control which slide they want to view.
4. Eventually we will move this out to the UI library, and its an option they can use in the future if they want to.